src/test/java/bio/terra/FileTypeDetectorTest.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra;
+
+import static org.junit.Assert.assertTrue;
+
+import bio.terra.cloudfunctions.common.FileTypeDetector;
+import bio.terra.cloudfunctions.common.GsonWrapper;
+import bio.terra.common.BaseTest;
+import com.google.events.cloud.storage.v1.StorageObjectData;
+import java.io.BufferedInputStream;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.junit.Test;
+
+public class FileTypeDetectorTest extends BaseTest {
+  @Test
+  public void gzipHandlerTest() {
+    try {
+      StorageObjectData storageObjectData =
+          GsonWrapper.getInstance().fromJson(MOCK_EVENT_GZIP, StorageObjectData.class);
+      FileTypeDetector fileTypeDetector = new FileTypeDetector(storageObjectData);
+      // Set a mock TGZ input stream to simulate the GCS input stream
+      fileTypeDetector.setInputStream(MOCK_TGZ);
+      ArchiveInputStream ais = (ArchiveInputStream) fileTypeDetector.getDataStream();
+      ArchiveEntry archiveEntry;
+      System.out.println("gzipHandlerTest:");
+      while ((archiveEntry = ais.getNextEntry()) != null) {
+        verifyMockTGZArchiveEntry(archiveEntry.getName(), archiveEntry.getSize());
+      }
+    } catch (Exception e) {
+    }
+  }
+
+  @Test
+  public void gzipHandlerTest2() {
```

Comment:
I think I see what this test is doing (compared to `gzipHandlerTest`), but can the test name(s) be updated to reflect that?

Reply:
Renamed the unit tests to more meaningful names.